File: Stage13/Python/Problem10814.py
import sys

def main():
    N = int(sys.stdin.readline())
    people = []
    
    for _ in range(N):
        age, name = sys.stdin.readline().split()
        people.append((int(age), name))
        
    people.sort(key=lambda x: x[0])
    
    for age, name in people:
        print(age, name)
    
if __name__ == "__main__":
    main()

File: Stage13/Java/Problem10814.java
package Stage13.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.StringTokenizer;

class Person {
    int age;
    String name;

    Person(int age, String name) {
        this.age = age;
        this.name = name;
    }
}

public class Problem10814 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        List<Person> list = new ArrayList<>();

        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int age = Integer.parseInt(st.nextToken());
            String name = st.nextToken();

            list.add(new Person(age, name));
        }

        Collections.sort(list, (a, b) -> a.age - b.age);

        for (Person p : list) {
            System.out.println(p.age + " " + p.name);
        }

        br.close();
    }
}
